File: source/exercises/saxpy.f90
program saxpy

  implicit none
  integer :: i,n
  real,dimension(:),allocatable :: x, y
  real :: a,start_time, end_time

  n=100000000
  allocate(x(n),y(n))
  ! Parallelize this block of code (optional)
  x = 1.0d0
  y = 2.0d0
  a = 2.0d0

  call cpu_time(start_time)
  ! Parallelize this block of code
  do i = 1, n
     y(i) = y(i) + a * x(i)
  end do
  call cpu_time(end_time)
  deallocate(x,y)
  
  print '(a,f8.6)', 'SAXPY Time: ', end_time - start_time
  
end program saxpy
    

File: source/exercises/saxpy.c
#include <stdio.h>
#include <time.h>

int main() {
  int i;
  long long int n=100000000;
  float a=2.0;
  float x[n];
  float y[n];
  clock_t start_time, end_time;

  /* Parallelize this block of code (optional) */
  for (i = 0; i < n; i++){
    x[i] = 1.0;
    y[i] = 2.0;
  }

  start_time = clock();
  /* Parallelize this block of code */
  for (i = 0; i < n; i++){
    y[i] = a*x[i] + y[i];
  }
  end_time = clock();
  printf ("SAXPY Time: %f\n",(double)(end_time - start_time)/CLOCKS_PER_SEC);
  
}
